signer/tests/integration/transaction_coordinator.rs

```diff
@@ -2228,3 +2228,50 @@ async fn test_get_btc_state_with_available_sweep_transactions_and_rbf() {
 
     testing::storage::drop_db(db).await;
 }
+
+
+async fn test_get_pending_requests_ignore_withdrawals() {
+    let mut rng = rand::rngs::StdRng::seed_from_u64(46);
+
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let db = testing::storage::new_test_database(db_num, true).await;
+
+    let client = BitcoinCoreClient::new(
+        "http://localhost:18443",
+        regtest::BITCOIN_CORE_RPC_USERNAME.to_string(),
+        regtest::BITCOIN_CORE_RPC_PASSWORD.to_string(),
+    )
+    .unwrap();
+
+    let context = TestContext::builder()
+        .with_storage(db.clone())
+        .with_bitcoin_client(client.clone())
+        .with_mocked_emily_client()
+        .with_mocked_stacks_client()
+        .build();
+    let network = SignerNetwork::single(&context);
+
+    let mut coord = TxCoordinatorEventLoop {
+        context,
+        private_key: PrivateKey::new(&mut rng),
+        network: network.spawn(),
+        threshold: 5,
+        context_window: 5,
+        signing_round_max_duration: std::time::Duration::from_secs(5),
+        dkg_max_duration: std::time::Duration::from_secs(5),
+        sbtc_contracts_deployed: false,
+        is_epoch3: true,
+        pre_sign_pause: Some(Duration::from_secs(1)),
+    };
+
+    let aggregate_key = &PublicKey::from_private_key(&PrivateKey::new(&mut rng));
+
+
+    let tx_coordinator_handle = tokio::spawn(async move { coord.run().await });
+
+
+    // TODO: hmm, how can I send transactions to this test network...
```

Comment:
This may not need to be an integration test, I suggest looking at the tests at the bottom of `/src/transaction_coordinator.rs` which use the coordinator `TestEnvironment` in `/src/testing/transaction_coordinator.rs` (where all of the actual logic lives). That uses an in-mem db, but if you follow the (albeit confusing) code, in the `test_environment()` method it passes the following parameters, which sets up some withdrawal requests:

```
let test_model_parameters = testing::storage::model::Params {
    num_bitcoin_blocks: 20,
    num_stacks_blocks_per_bitcoin_block: 3,
    num_deposit_requests_per_block: 5,
    num_withdraw_requests_per_block: 5,
    num_signers_per_request: 7,
};
```